### ISSUE TITLE
Loading Data with DALI results in Protobuf String error

### ISSUE THREAD

**Issue by LucasSloan** [NVIDIA/DALI#2699]
Tensorflow version: 2.4.1
Cuda version: 11
OS: Ubuntu 20.04

I've constructed a DALI pipeline that takes in tfrecords and is consumed as a tf.Dataset.  When I run it with my code, I get the following error:

```
python3: ./google/protobuf/stubs/stringpiece.h:229: google::protobuf::StringPiece::StringPiece(const char*, google::protobuf::stringpiece_ssize_type): Assertion `len >= 0' failed.
```

I initially thought the error was in unpacking the jpeg data out of my tfrecords (stored as `tf.io.FixedLenFeature([], tf.string)`), but modifying my pipeline to not decode those fields doesn't fix the issue.

I've also tried fooling around with the data format (reducing the number of label types, etc.) but that doesn't seem to work.

My code is [here](https://github.com/LucasSloan/speedchallenge/blob/master/dali_loader.py) and it's consumed in this [file](https://github.com/LucasSloan/speedchallenge/blob/master/keras_predictor.py#L22).

**Comment by LucasSloan** [comment]
Figured it out.  The `path` and `index_path` arguments to `fn.tfrecord_reader` need to be parallel.  Is it possible to give this a better error message?

**Comment by JanuszL** [comment]
Hi @LucasSloan,

>  The path and index_path arguments to fn.tfrecord_reader need to be parallel. Is it possible to give this a better error message?

If you provide completely random files to `index_path` DALI cannot detect that they are not aligned with the TFRecord files. `index_path` points to files with the list of offsets to the records inside TFRecord. If these offsets are wrong DALI will read random parts of the file and protobuf will fail to parse them. I don't have any good idea how to detect such a situation.

**Comment by LucasSloan** [comment]
Could we change the format of the .idx files to start with the file name of the file they refer to?

**Comment by JanuszL** [comment]
> Could we change the format of the .idx files to start with the file name of the file they refer to?

Would you mind posting a PR with the proposed functionality?
